### PR TITLE
Laushinka/let people unsubscribe 4761

### LINKED ISSUE #4761: Let people unsubscribe from emails without logging in
**Problem:**
- Currently, users need to login in before there are able to unsubscribe from emails. The unsubscribe link links to https://gitpod.io/notifications. Legally we must provide a way to unsubscribe within a single page [[1](https://uxdesign.cc/log-in-to-unsubscribe-ea0b9e99be1e)]. 
- With the new [newsletter widget](**https://www.gitpod.io/blog#newsletter**), people that don't have a Gitpod account can also sign up for emails. They need to have an option to unsubscribe from emails as well. 

**Solution:**
Creating an API where people can unsubscribe without logging in

**Acceptance criteria:**
- [x] Create an "Unsubscribed successful" page on website
- [x] Create a new API on the server
- [ ] Change links in the email campaigns to redirect to the new API

**Note:** changes in /notification are already automatically forwarded and updated in customer.io

**Context:**
[customer.io docs](https://customer.io/docs/multiple-subscription-types/)
[internal discussion](https://gitpod.slack.com/archives/C0274BHF9UM/p1625811273000600?thread_ts=1625735873.006900&cid=C0274BHF9UM)

cc @jakobhero @atduarte

### PR DESCRIPTION
**Pull request by laushinka**
This PR created initially to run the following command and test that the new events are reflected on Segment.

/werft analytics=segment|TEZnsG4QbLSxLfHfNieLYGF4cDwyFWoe

Fixes #4761

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#4761*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by JanKoehnlein** [comment]
Product question: Should this be an Enterprise Edition feature?

**Comment by laushinka** [comment]
> Product question: Should this be an Enterprise Edition feature?

Who should answer this?

**Comment by JanKoehnlein** [comment]
| Who should answer this?
Somebody from our product team. I put it in their inbox (Product Groundwork project) so they should answer fast

File: components/server/src/user/newsletter-subscription-controller.ts
Comment (jakobhero) [inline components/server/src/user/newsletter-subscription-controller.ts:57]: Let's also change this to an `identify` call in order to be consistent with what 1) we are doing in the else branch of the conditional and 2) customer.io / other 3rd party software is expecting downstream [(see here)](https://customer.io/docs/segment-destination/).

File: components/server/src/user/newsletter-subscription-controller.ts
Comment (jakobhero) [inline components/server/src/user/newsletter-subscription-controller.ts:20]: we would like to use this API to subscribe newsletter users so that they are automatically attached to their user accounts if a the email for the newsletter signup matches any user's primary email. is it ok for you to extend the scope of this PR to include a `/subscribe` endpoint that signs up a known or unknown user for a specified subscription list?

**Review by jakobhero — CHANGES_REQUESTED** [review]
Please use `identify` instead of `notification_change` track call and - if ok to include in the scope of this PR - add a `/subscribe` endpoint that subscribes a user to the passed mailing list.

File: components/server/src/user/newsletter-subscription-controller.ts
Comment (laushinka) [reply components/server/src/user/newsletter-subscription-controller.ts:57]: Will do.

File: components/server/src/user/newsletter-subscription-controller.ts
Comment (laushinka) [reply components/server/src/user/newsletter-subscription-controller.ts:20]: I would prefer to keep the existing scope of this ticket for reasons of: 1) planning and 2) transparency :) When tickets are approved and put to in progress, I assume there are already plans and estimations of how much effort the current ticket/board would take. Changing the scope might change the planning for the rest of the board. Secondly, a changing scope almost at the end of the work might not be known to the stakeholders of the board.
However, I am still learning how the team usually manages their projects so I will tag @JanKoehnlein here for his thoughts.

**Comment by svenefftinge** [comment]
> Product question: Should this be an Enterprise Edition feature?

It can stay in the open-source code base. We would only consider code that adds useful enterprise features to be moved to an enterprise-licensed directory, everything else goes open-source.

File: components/server/src/user/newsletter-subscription-controller.ts
Comment (JanKoehnlein) [reply components/server/src/user/newsletter-subscription-controller.ts:20]: Let's open another ticket for @jakobhero's request. I feel like it has to be discussed in terms of security and GDPR as well.

**Review by JanKoehnlein — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: fb94e55211835b3e7496627d88980e9742c4542a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5281#pullrequestreview-736986164" title="Approved">JanKoehnlein</a>*

Associated issue: *#4761*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
